MushroomObserver/mushroom-observer
# New AJAX "Propose a Name" popup form for show_obs page

> Depends on #1345, and includes those changes. # of Files changed will be lower when that merges.
Testable separately. 

This PR adds a new UI for proposing a naming for an observation. It's useful for the show_obs page, but also intended for the upcoming "Observations Needing Identifications" PR.  It makes a "new naming form" _modal_ (i.e., a Bootstrap "popup") that can be called from any context on an observation.

## Please test this manually ##

**Improved functionality:** 
- You can now propose a naming without leaving the show_obs page. Autocompletes and form errors are handled within the modal, just as they would be on the "propose naming" form. On successful submit, the naming form in the underlying page will update with your new naming, adjusted votes, and the title of the obs will update if appropriate, all without a page reload. 

**Coding notes:**
- The refactor uses `respond_to do |format|` to route the request for different formats, and follows the same pattern as the previous "Naming Vote via AJAX" PR. That means the same controller action can respond to HTML and AJAX requests. 
- JS requests to an action look for a JS template file named after the actions, which are `Observations::NamingsController#new` and `#create` (note that unlike HTML templates, there's a separate JS file called on create, as there needs to be.)
  - the form is displayed in the modal (and bindings added) by `new.js.erb`
  - the live update (to the show_obs template) is handled by a new template file, `create.js.erb`
  - the form reload and JS bindings are handled by `form_reload.js.erb`

**Manual tests:** 
- On an observation, click `Propose another name`. Propose a valid naming. The naming should be added to the observation
- Propose an already-proposed naming
- Propose deprecated names
- Try doing the same with JS turned off in the browser. The propose naming form should load in a new page as usual. (It doesn't break existing functionality).
